app/uk/gov/hmrc/eeittadminfrontend/controllers/GformsController.scala

```diff
@@ -16,10 +16,80 @@
 
 package uk.gov.hmrc.eeittadminfrontend.controllers
 
+import play.api.data.Form
+import play.api.data.Forms._
 import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.Json
+import uk.gov.hmrc.eeittadminfrontend.AppConfig
+import uk.gov.hmrc.eeittadminfrontend.config.Authentication
+import uk.gov.hmrc.eeittadminfrontend.connectors.GformConnector
+import uk.gov.hmrc.eeittadminfrontend.models.{FormTypeId, GformIdAndVersion}
 import uk.gov.hmrc.play.frontend.auth.Actions
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
 import uk.gov.hmrc.play.frontend.controller.FrontendController
 
-class GformsController(val authConnector: AuthConnector, val messagesApi: MessagesApi) extends FrontendController with Actions with I18nSupport
+import scala.concurrent.Future
+
+class GformsController(val authConnector: AuthConnector)(implicit appConfig: AppConfig, val messagesApi: MessagesApi) extends FrontendController with Actions with I18nSupport {
```

Comment:
1. `val` exposes `authConnector`. Since we care about _encapsulation_ we should not use it here.
2. Implicit parameters should be palces on _normal_ argument list.